code/modules/jobs/departments/departments.dm

```diff
@@ -28,14 +28,15 @@
 	/// A list of generic access flags people in this department generally have.
 	var/list/department_access = list()
 
-/datum/job_department/New()
+/datum/job_department/New(check_areas = TRUE)
 	. = ..()
-	for(var/delivery_area_type in department_delivery_areas)
-		if(GLOB.areas_by_type[delivery_area_type])
-			return
-	//every area fallback didn't exist on this map so throw a mapping error and set some generic area that uuuh please exist okay
-	log_mapping("[type] has no valid areas to deliver to on this map, add some more fallback areas to its \"department_delivery_areas\" var.")
-	department_delivery_areas = list(/area/station/hallway/primary/central) //if this doesn't exist like honestly fuck your map man
+	// Speeds up any future loops
+	if(check_areas == TRUE)
+		for(var/delivery_area_type in department_delivery_areas)
+			if(GLOB.areas_by_type[delivery_area_type])
+				department_delivery_areas = list(delivery_area_type)
+				return
+		department_delivery_areas = list(/area/station/hallway/primary/central) //if this doesn't exist like honestly fuck your map man
```

Comment:
We don't even need any code in `/datum/job_department/New` anymore, the unit test can handle this entirely